.github/workflows/lint-and-test.yml

```diff
@@ -136,6 +136,23 @@ jobs:
       - name: Run clippy::pedantic for proof-of-sql-parser
         run: cargo clippy --lib -p proof-of-sql-parser -- -D clippy::pedantic
 
+  coverage:
+    name: Code Coverage
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        features: [ "test", "blitzar", "arrow", "rayon", "perf", "std", "default" ]
```

Comment:
Since many of these are part of the `default` feature, and you aren't using `--no-default-features`, most aren't really doing anything. Also, some depend on others, and we probably don't need to isolate each one.

These are the combinations that matter and should run all combinations of tests and features:
```
--all-features
--no-default-features --features="rayon"
--no-default-features --features="blitzar"
--no-default-features
```

Reply:
Also, I saw this: https://github.com/taiki-e/cargo-llvm-cov?tab=readme-ov-file#merge-coverages-generated-under-different-test-conditions.

Looks like it's possible to have a single report for all coverage rather than multiple reports.